framework/src/utils/ImageSampler.C

```diff
@@ -26,8 +26,8 @@ InputParameters validParams<ImageSampler>()
 
   params.addParam<Point>("origin", "Origin of the image (defaults to mesh origin)");
   params.addParam<Point>("dimensions", "x,y,z dimensions of the image (defaults to mesh dimensions)");
-  params.addParam<unsigned int>("component", "The image component to return, leaving this blank will result in a greyscale value "
-                                             "for the image to be created. The component number is zero based, i.e. 0 returns the first component of the image");
+  params.addParam<unsigned int>("component", "The image RGB-component to return, leaving this blank will result in a greyscale value "
```

Comment:
Perhaps I'll just RGB as an example. It's confusing to talk about component
of an image...
On Mon, Jul 18, 2016 at 5:03 PM Daniel Schwen <notifications@github.com>
wrote:

> In framework/src/utils/ImageSampler.C
> <https://github.com/idaholab/moose/pull/7352#discussion_r71247011>:
>
> > @@ -26,8 +26,8 @@ InputParameters validParams<ImageSampler>()
> >
> >    params.addParam<Point>("origin", "Origin of the image (defaults to mesh origin)");
> >    params.addParam<Point>("dimensions", "x,y,z dimensions of the image (defaults to mesh dimensions)");
> > -  params.addParam<unsigned int>("component", "The image component to return, leaving this blank will result in a greyscale value "
> > -                                             "for the image to be created. The component number is zero based, i.e. 0 returns the first component of the image");
> > +  params.addParam<unsigned int>("component", "The image RGB-component to return, leaving this blank will result in a greyscale value "
>
> Yeah, nevermind. The number of components is determined for each image
> individually with _data->GetNumberOfScalarComponents(); so an enum
> wouldn't make sense here (but neither does being overly specific in the
> doco string, as you cannot guarantee what the components in the image
> really are).
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/idaholab/moose/pull/7352/files/4684d44a3d766f924507f12e7db567313779ca75#r71247011>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AC5XIDkIWR7t_5z16mVtAwSycxEUTWJWks5qXAYngaJpZM4JO8NB>
> .
>